packages/terminal/src/browser/terminal-frontend-contribution.ts

```diff
@@ -68,12 +80,34 @@ export class TerminalFrontendContribution implements TerminalService, CommandCon
             isEnabled: () => this.shell.activeWidget instanceof TerminalWidget,
             execute: () => (this.shell.activeWidget as TerminalWidget).clearOutput()
         });
+
+        commands.registerCommand(TerminalCommands.TERMINAL_CONTEXT, new UriAwareCommandHandler<URI>(this.selectionService, {
+            execute: async uri => {
+                // Determine folder path of URI
+                let cwd = normalize(uri.path.toString());
+                const stat = await this.fileSystem.getFileStat(cwd);
```

Comment:
only URIs should be passed between frontend and backend, i.e. `this.fileSystem.getFileStat(uri)`